specification/sections/01-flag-evaluation.md

```diff
@@ -163,66 +236,84 @@ FlagEvaluationDetails<MyStruct> myStructDetails = client.getObjectDetails<MyStru
 
-##### Requirement 1.4.7
+#### Requirement 1.4.8
 
 > In cases of abnormal execution, the `evaluation details` structure's `error code` field **MUST** contain an `error code`.
 
 See [error code](../types.md#error-code) for details.
 
-##### Requirement 1.4.8
+#### Requirement 1.4.9
 
 > In cases of abnormal execution (network failure, unhandled error, etc) the `reason` field in the `evaluation details` **SHOULD** indicate an error.
 
-##### Requirement 1.4.9
+#### Requirement 1.4.10
 
 > Methods, functions, or operations on the client **MUST NOT** throw exceptions, or otherwise abnormally terminate. Flag evaluation calls must always return the `default value` in the event of abnormal execution. Exceptions include functions or methods for the purposes for configuration or setup.
 
 Configuration code includes code to set the provider, instantiate providers, and configure the global API object.
 
-##### Requirement 1.4.10
+#### Requirement 1.4.11
 
 > In the case of abnormal execution, the client **SHOULD** log an informative error message.
 
 Implementations may define a standard logging interface that can be supplied as an optional argument to the client creation function, which may wrap standard logging functionality of the implementation language.
 
-##### Requirement 1.4.11
+#### Requirement 1.4.12
 
 > The `client` **SHOULD** provide asynchronous or non-blocking mechanisms for flag evaluation.
 
 It's recommended to provide non-blocking mechanisms for flag evaluation, particularly in languages or environments wherein there's a single thread of execution.
 
-##### Requirement 1.4.12
+#### Requirement 1.4.13
 
 > In cases of abnormal execution, the `evaluation details` structure's `error message` field **MAY** contain a string containing additional details about the nature of the error.
 
-#### Evaluation Options
+### Evaluation Options
 
-##### Requirement 1.5.1
+#### Requirement 1.5.1
 
 > The `evaluation options` structure's `hooks` field denotes an ordered collection of hooks that the client **MUST** execute for the respective flag evaluation, in addition to those already configured.
 
 See [hooks](./04-hooks.md) for details.
+
+### 1.6. Shutdown
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 1.6.1
+
+> The API **MUST** define a `shutdown` or `dispose` function, which, when called, must call the respective `shutdown`/`dispose` method on the active provider.
+
+see: [`shutdown`/`dispose`](./02-providers.md#24-shutdown)
+
+#### Requirement 1.6.2
+
+> When the API's `provider mutator` is invoked, the `shutdown`/`dispose` method on the active provider **MUST** run, if defined.
+
+Setting a new provider means the previous provider is no longer in use, and should therefor be disposed of to prevent resource leaks.
```

Comment:
Do we want to be more clear in which circumstances this would be the case? So there is a bit of a checklist implementators can check against? Instead of this vague sentence?

Reply:
I will add a some more detail, and an link to disposal.